GG/GG/Font.h

```diff
@@ -147,6 +151,13 @@ class GG_API Font
             pair.second. */
         Substring(const std::string& str_, const IterPair& pair);
 
+        /** Attach this Substring to \p str_.
+
+            This changes the iterators from pointing into the previous
```

Comment:
Rebinding a SubString doesn't invalidate in the same sense as the STL, though. You could have several substrings pointing into the same std::string, and rebinding one would not affect the others, because the pointed-to string is still valid and unchanged by the substring rebinding.